### PR TITLE
Add MistralAI Embeddings

### PR DESCRIPTION
**Pull request by ravi03071991**
# Description

PR to add Mistral AI Embeddings.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/embeddings/mistralai.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    Args:
+        model_name (str): Model for embedding.
+            Defaults to "mistral-embed".
+
+        api_key (Optional[str]): API key to access the model. Defaults to None.
+    """
+
+    # Instance variables initialized via Pydantic's mechanism
+    mistralai_client: Any = Field(description="MistralAI client")
```
Comment (logan-markewich) [inline llama_index/embeddings/mistralai.py:21]: Can we make this `_mistralai_client: Any = PrivateAttr()`

and then in the init

```
        self._mistralai_client = MistralClient(api_key=api_key),
        super().__init__(
            model_name=model_name,
            embed_batch_size=embed_batch_size,
            callback_manager=callback_manager,
            **kwargs,
        )
```

File: llama_index/embeddings/mistralai.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        """Get query embedding."""
+        return (
+            self.mistralai_client.embeddings(model=self.model_name, input=[query])
+            .data[0]
+            .embedding
+        )
+
+    async def _aget_query_embedding(self, query: str) -> List[float]:
+        """The asynchronous version of _get_query_embedding."""
+        return self._get_query_embedding(query)
```
Comment (logan-markewich) [inline llama_index/embeddings/mistralai.py:59]: I guess the client doesn't have async yet?

Comment (ravi03071991) [reply llama_index/embeddings/mistralai.py:59]: Yes.

File: llama_index/embeddings/mistralai.py (hunk 1 above)
Comment (ravi03071991) [reply llama_index/embeddings/mistralai.py:21]: Yes. Sure.

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**